### PR TITLE
MGMT-11271: Add Nutanix Platform type

### PR DESCRIPTION
**Pull request by vrutkovs**
Add Nutanix support, detection and a unittests which ensures that a stub install config is generated. Example install config: https://github.com/openshift/release/blob/master/ci-operator/step-registry/ipi/conf/nutanix/ipi-conf-nutanix-commands.sh#L30-L46=

This also adds a new feature support level (unsupported until 4.11, dev preview in 4.11 and 4.12)

## List all the issues related to this PR

- [x] New Feature 

## What environments does this code impact?

- [x] Automation (CI, tools, etc)
- [x] Cloud

## How was this code tested?

- [x] Manual (Elaborate on how it was tested)

## Assignees

/cc @michaellevy101
/cc @gamli75

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by gamli75** [comment]
@eliorerz please review

File: internal/installcfg/installcfg.go [hunk 1]
```diff
@@ -43,6 +44,37 @@ type VsphereInstallConfigPlatform struct {
 	IngressVIP       string          `yaml:"ingressVIP"`
 }
 
+type NutanixInstallConfigPlatform struct {
+	ID            int                   `yaml:"-"`
+	APIVIP        string                `yaml:"apiVIP"`
```
Comment (romfreiman) [inline internal/installcfg/installcfg.go:49]: think about UMN as well - then APIVIP/IngressVIP wont be required.

File: internal/provider/nutanix/consts.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package nutanix
+
+const (
+	PhUsername   = "username_placeholder"
```
Comment (romfreiman) [inline internal/provider/nutanix/consts.go:4]: please align with the vSphere palceholder namings

File: internal/provider/nutanix/installConfig.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+		},
+	}
+	platform.SubnetUUIDs = []strfmt.UUID{
+		strfmt.UUID(PhSubnetUUID),
+	}
+}
+
+func (p nutanixProvider) AddPlatformToInstallConfig(
+	cfg *installcfg.InstallerConfigBaremetal, cluster *common.Cluster) error {
+	if len(cluster.APIVip) == 0 {
```
Comment (romfreiman) [inline internal/provider/nutanix/installConfig.go:37]: think about UMN as well here

**Comment by romfreiman** [comment]
Few comments:
1. I assume we'll have to enable it starting 4.11 (there is no ntnx support in 4.10 and earlier afaik)
2. UMN is also a requirement. We might want to block it in the first step, but later on it will be availalbe
3. Will we be able to identify the installation type in ELK (I assume we do)
4. Add feature usage

**Comment by osherdp** [comment]
/hold
just until we release AI 2.7.0

**Comment by vrutkovs** [comment]
AI 2.7.0 released, cancelling the hold

/hold cancel

**Comment by vrutkovs** [comment]
Possibly flakes
/retest

File: internal/provider/nutanix/installConfig.go (hunk 3 above)
Comment (eliorerz) [reply internal/provider/nutanix/installConfig.go:38]: Do we wan't to return error when UMN is enabled?

Comment (eliorerz) [reply internal/provider/nutanix/installConfig.go:38]: If no we might want to to something similar to vsphere:

```go
	if !swag.BoolValue(cluster.UserManagedNetworking) {
		if len(cluster.APIVip) == 0 {
			return errors.New("invalid cluster parameters, APIVip must be provided")
		}

		if len(cluster.IngressVip) == 0 {
			return errors.New("invalid cluster parameters, IngressVip must be provided")
		}

		vsPlatform.APIVIP = cluster.APIVip
		vsPlatform.IngressVIP = cluster.IngressVip
	} else {
		cfg.Networking.MachineNetwork = provider.GetMachineNetworkForUserManagedNetworking(p.Log, cluster)
		if cluster.NetworkType != nil {
			cfg.Networking.NetworkType = swag.StringValue(cluster.NetworkType)
		}
	}


```

File: internal/provider/nutanix/inventory.go [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+	"github.com/openshift/assisted-service/models"
+)
+
+func (p *nutanixProvider) CleanPlatformValuesFromDBUpdates(_ map[string]interface{}) error {
+	return nil
+}
+
+func (p *nutanixProvider) SetPlatformValuesInDBUpdates(_ *models.Platform, _ map[string]interface{}) error {
+	return nil
+}
```
Comment (eliorerz) [inline internal/provider/nutanix/inventory.go:14]: This method is no longer exists

Comment (vrutkovs) [reply internal/provider/nutanix/inventory.go:14]: Removed, thanks

File: internal/provider/nutanix/installConfig.go (hunk 3 above)
Comment (vrutkovs) [reply internal/provider/nutanix/installConfig.go]: Lets make it similar to vSphere and run another round of tests. If UMN is confirmed to be impossible on Nutanix we'll disable it entirely

File: internal/featuresupport/support_levels_list.go [hunk 5]
```diff
@@ -262,12 +286,16 @@ var SupportLevelsList = models.FeatureSupportLevels{
 				FeatureID:    models.FeatureSupportLevelFeaturesItems0FeatureIDVIPAUTOALLOC,
 				SupportLevel: models.FeatureSupportLevelFeaturesItems0SupportLevelDevPreview,
 			},
+			{
+				FeatureID:    models.FeatureSupportLevelFeaturesItems0FeatureIDNUTANIXINTEGRATION,
+				SupportLevel: models.FeatureSupportLevelFeaturesItems0SupportLevelUnsupported,
```
Comment (gamli75) [inline internal/featuresupport/support_levels_list.go:291]: why it's unsupported in 4.12? it should be FeatureSupportLevelFeaturesItems0SupportLevelSupported

File: internal/featuresupport/support_levels_list.go [hunk 6]
```diff
@@ -221,6 +241,10 @@ var SupportLevelsList = models.FeatureSupportLevels{
 				FeatureID:    models.FeatureSupportLevelFeaturesItems0FeatureIDLVM,
 				SupportLevel: models.FeatureSupportLevelFeaturesItems0SupportLevelDevPreview,
 			},
+			{
+				FeatureID:    models.FeatureSupportLevelFeaturesItems0FeatureIDNUTANIXINTEGRATION,
+				SupportLevel: models.FeatureSupportLevelFeaturesItems0SupportLevelUnsupported,
```
Comment (gamli75) [inline internal/featuresupport/support_levels_list.go:246]: should be FeatureSupportLevelFeaturesItems0SupportLevelTechPreview  in 4.11

Comment (vrutkovs) [reply internal/featuresupport/support_levels_list.go:246]: Fixed

File: internal/featuresupport/support_levels_list.go (hunk 5 above)
Comment (vrutkovs) [reply internal/featuresupport/support_levels_list.go:291]: Right, moved this to "Supported"

**Review by eliorerz — APPROVED** [review]
LGTM

File: internal/featuresupport/support_levels_list.go (hunk 6 above)
Comment (vrutkovs) [reply internal/featuresupport/support_levels_list.go]: Right, updated

File: internal/featuresupport/support_levels_list.go (hunk 5 above)
Comment (vrutkovs) [reply internal/featuresupport/support_levels_list.go]: Fixed, moved to supported

**Comment by vrutkovs** [comment]
```
 Assisted service ci template does not exist
INFO[2022-09-20T14:40:32Z] Step e2e-vsphere-assisted-assisted-vsphere-setup-machine failed after 2m0s. 
```

Not sure if its flake or a permafailing problem
/test edge-e2e-vsphere-assisted

File: internal/provider/nutanix/installConfig.go (hunk 3 above)
Comment (romfreiman) [reply internal/provider/nutanix/installConfig.go]: for some reason, i recalled that @rvanderp3 said it should work, just not tested. But i might be wrong here

**Comment by vrutkovs** [comment]
Flakey flakes
/retest

**Review by gamli75 — APPROVED** [review]

**Pull request merged**